### PR TITLE
Lockable schoolgirl stuff

### PR DESCRIPTION
**Pull request by MystLeissa**
## About The Pull Request

Adds a set of lockable schoolgirl uniforms along with a key to unlock them. They can be purchased from the supply terminal and contain a full set of colors including the new PINK uniform which ONLY has a locking variant. The crate also comes with a lockbox containing the key

Lockable Schoolgirl uniforms are a double edged sword, on the one hand they won't be coming off easily, and prevent you from wearing outerwear. On the other they feature 4 Extra "Hidden" Stealth/Silent slots compared to normal uniforms (accessed by alt+clicking on the icon while worn).

The Lockbox containing the key requires Captain-class Access Perms to open

## Why It's Good For The Game

Variety, Fun, and a undoubtly future source of meme's can't be bad right? In all seriousness - this is a "fun" update - but one I take pride in.

## Changelog
:cl:
add: Lockable Schoolgirl Uniforms
balance: Now you can't be sure that pretty uniform is a harmless prank
imageadd: Pink School Girl Uniform
spellcheck: Spellcheck? Not doing it - you can; All I Know is it works.
refactor: Refactored _uniforms.dm, _spacesuits.dm, and _suits.dm to allow for unique TRAIT_NOSUIT effect (trait used by lockable uniforms to prevent wearing outerwear
code: Moved Schoolgirl uniforms code to dedicated file)
server: I take no responsibility for any claims of weirdness arising from this patch - You are on your own.
server: Also there seems to be a problem with the map version I'm submitting the PR anyway because I don't know what's causing it but there SHOULD be no map changes in this PR so I don't know why it's throwing that at me in the server console. Seems to work regardless.
/:cl:

### CODE REVIEW

**Comment by bgobandit** [comment]
it's truly remarkable how many different kinds of fetish content this adds in one PR

:-1:

**Comment by 81Denton** [comment]
Lockable school girl outfits don't really fit the theme of /tg/station, but I think that this PR would be well-received on codebases like Citadel or Paradise.

**Comment by MystLeissa** [comment]
> it's truly remarkable how many different kinds of fetish content this adds in one PR
> 
Well I guess you could LOOK at it that way. but It's not intended to be a Fetish Patch - it's more fun-oriented. It's more an outfit update with perks and balances - Plus I felt we were missing pink uniforms - if I /have/ to I'll remove the lockable content and re-PR the pink uniform on it's own i think - Pink is my favorite color and it sucks that there's no pink uniform in the base game...

**Comment by Repukan** [comment]
I read that doujin too, yeah

**Comment by MystLeissa** [comment]
> I read that doujin too, yeah

what's doujin?

**Comment by Akrilla** [comment]
doujin pr tags when

**Comment by mrhugo13** [comment]
Thank fuck now i can finally RP as a 6 year old japanese schoolgirl in my 2d spaceman simulator

**Comment by nervere** [comment]
This is, without any shadow of a doubt, fetish content.
Your comment in the changelog (why the changelog?) reaffirms this:
>server: I take no responsibility for any claims of weirdness arising from this patch - You are on your own.

Please PR this to a downstream that allows for ERP, as we do not want this here.

**Comment by MystLeissa** [comment]
> This is, without any shadow of a doubt, fetish content.
> Your comment in the changelog (why the changelog?) reaffirms this:
> 
> > server: I take no responsibility for any claims of weirdness arising from this patch - You are on your own.
> 
> Please PR this to a downstream that allows for ERP, as we do not want this here.

That was a joke..It's seriously not fetish content - what about all that mumbo-jumbo about Necropolis "curses" - is that fetish content too? I mean really - there's a million and one ways to frame it. I just chose one that was easy to think of in my mind.

File: code/modules/clothing/under/schoolgirl.dm [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+/obj/item/clothing/under/schoolgirl/locked/attackby(obj/item/I, mob/user, params)
+	/*if(perma)
+		return
+	if(I == /obj/item/locks/perma)
+		to_chat(user,"You swap out the lock for a permanent type")
+		perma = TRUE
+		return
+	if(I == /obj/item/locks/slave)
+		to_chat(user,"You swap out the standard lock for a slave lock")
```
Comment (Shapsy) [inline code/modules/clothing/under/schoolgirl.dm]: uh oh

Comment (MystLeissa) [reply code/modules/clothing/under/schoolgirl.dm]: I really need to remove my comments sometime - that was a failed project...it didn't work out well...

File: code/modules/clothing/under/schoolgirl.dm [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	to_chat(user,"It's locked.")
+
+/obj/item/clothing/under/schoolgirl/locked/examine(mob/user)
+
+	if(locked)
+		. += "<span class='notice'>It's locked..</span>"
+	if(perma)
+		. += "<span class='notice'>It's impossible to unlock.</span>"
+	if(slaved)
+		. += "<span class-'notice'>THey belong to [slaved_to].</span>"
```
Comment (81Denton) [inline code/modules/clothing/under/schoolgirl.dm]: hmm

**Comment by nervere** [comment]
Your account is brand-new and is immediately trying to make controversial, gross changes.
This seems like a bad faith attempt from another server to cause turmoil within our community.

**Comment by 81Denton** [comment]
For the sake of everyone's dignity, I'm gonna close this one.

**Comment by MystLeissa** [comment]
> Your account is brand-new and is immediately trying to make controversial, gross changes.
> This seems like a bad faith attempt from another server to cause turmoil within our community.

Would you prefer we resolved this as adults or children? Because I can think of a few things to say about your "faith" as well but I'm not going to elaborate.

**Comment by Screemonster** [comment]
put me in the screenshot

**Comment by qbmax32** [comment]
cringed and yikespilled

**Comment by optimumtact** [comment]
brand new? it was made in 2017

**Comment by sustox** [comment]
the absolute state of /tg/

**Comment by Shadowflame909** [comment]
Johnny Bravo

**Comment by PowerfulBacon** [comment]
wtf

**Comment by amidoingitright** [comment]
what the hell happened here?

**Pull request closed without merging**